appengine/endpoints-frameworks-v2/backend/src/main/java/com/example/echo/Echo.java

```diff
@@ -59,19 +125,48 @@ public Message echo(Message message) {
    * Note that httpMethod is not required here. Without httpMethod, this will default to GET due
    * to the API method name. httpMethod is added here for example purposes.
    */
+  // [START google_id_token_auth]
   @ApiMethod(
       httpMethod = ApiMethod.HttpMethod.GET,
       authenticators = {EspAuthenticator.class},
-      audiences = {"YOUR_OAUTH_CLIENT_ID"},
-      authLevel = AuthLevel.REQUIRED
+      audiences = {"YOUR_OAUTH_CLIENT_ID"}
       )
   public Email getUserEmail(User user) throws UnauthorizedException {
     if (user == null) {
-      throw new UnauthorizedException();
+      throw new UnauthorizedException("Invalid credentials");
+    }
+
+    Email response = new Email();
+    response.setEmail(user.getEmail());
+    return response;
+  }
+  // [END google_id_token_auth]
+
+  /**
+   * Gets the authenticated user's email. If the user is not authenticated, this will return an HTTP
+   * 401.
+   *
+   * Note that name is not specified. This will default to "{class name}.{method name}". For
+   * example, the default is "echo.getUserEmail".
+   *
+   * Note that httpMethod is not required here. Without httpMethod, this will default to GET due
+   * to the API method name. httpMethod is added here for example purposes.
+   */
+  // [START firebase_auth]
+  @ApiMethod(
+      path = "firebase_user",
+      httpMethod = ApiMethod.HttpMethod.GET,
+      authenticators = {EspAuthenticator.class},
+      issuerAudiences = {@ApiIssuerAudience(name = "firebase", audiences = {"tangd-customer"})}
```

Comment:
same question - is `tangd-customer` something that should be modified by the user?  If so, a comment would be good.